partd/pandas.py

```diff
@@ -109,22 +114,33 @@ def index_from_header_bytes(header, bytes):
 def block_to_header_bytes(block):
     values = block.values
     if isinstance(values, pd.Categorical):
-        cat = (values.ordered, values.categories)
+        extension = (_EXTENSION_CATEGORICAL, (values.ordered,
+                                              values.categories))
         values = values.codes
+    elif pd.api.types.is_datetime64tz_dtype(block):
+        # TODO: compat with older pandas?
```

Comment:
If convenient I recommend that we just raise an informative error.  I don't mind nudging users towards newer libraries.